src/adapter/src/coord/dataflows.rs

```diff
@@ -548,15 +555,16 @@ pub fn prep_scalar_expr(
         ExprPrepStyle::OneShot {
             logical_time,
             session,
+            catalog_state,
         } => expr.try_visit_mut_post(&mut |e| {
             if let MirScalarExpr::CallUnmaterializable(f) = e {
-                *e = eval_unmaterializable_func(state, f, logical_time, session)?;
+                *e = eval_unmaterializable_func(catalog_state, f, logical_time, session)?;
             }
             Ok(())
         }),
 
         // Reject the query if it contains any unmaterializable function calls.
-        ExprPrepStyle::Index | ExprPrepStyle::AsOfUpTo => {
+        ExprPrepStyle::Index { .. } | ExprPrepStyle::AsOfUpTo => {
```

Comment:
```suggestion
        ExprPrepStyle::Index | ExprPrepStyle::AsOfUpTo => {
```

Do you have some commit skew? Index in the enum above doesn't have fields.

Reply:
Interesting! While I was working on this I added `catalog_state` to the `Index` variant, which required me to add the `{ .. }`. I thought Rust would have rejected the curly braces since Index has no fields, but I guess not! I went through and removed all of them